vendor/intel/avcstreamoutdemo.c

```diff
@@ -0,0 +1,1436 @@
+    0xa6fe32ce, 0xc76f64e6, 0xf72b760b, 0x32b1ca76,
+    0xde807e6c, 0x0bc27b7f, 0xd14d0204, 0xe4d7f39d,
+    0xb2d40890, 0x2c7b0080, 0x01640a31, 0x7828377f,
+    0xd0b57923, 0x7a8e81ee, 0x1ea59e1d, 0xd32905e5,
+    0x51103cf9, 0x0822a87d, 0x9fc886b9, 0x27aaa526,
+    0xc4c25a0c, 0x39eff6d1, 0x42be2114, 0xb1e0be80,
+    0x71551a96, 0x65ec526e, 0x5254683a, 0x2f53a761,
+    0x817c79c5, 0x22605079, 0x7b32b62b, 0xf94d4576,
+    0xf83c836d, 0xf35ca5ae, 0x48385aa1, 0x03ec899f,
+    0xbf3cb34a, 0x28292c12, 0x2b322a70, 0x3a8920e1,
+    0x0d04313b, 0x92b96796, 0x2264d217, 0xa175f3dd,
+    0x477af378, 0xf3d7cc31, 0x742867ff, 0xe04c2a32,
+    0xbfbdf54c, 0x78c04b0c, 0x8887a4f1, 0x77b63003,
+    0x7d45a0df, 0x41ec95db, 0xa6508183, 0x58343aaa,
+    0xca8380c1, 0x8bda0563, 0x853a1b72, 0x0525d346,
+    0x9790cbe7, 0x30881bb6, 0xaa6fc729, 0xd36cfbd4,
+    0x8d76b7fa, 0x29a871d5, 0xdae527bd, 0x3030e020,
+    0x8bcf8ab7, 0x61cb2f97, 0x13846041, 0x4ecde00c,
+    0xdef0ff00, 0xb71fa71b, 0xec2f31be, 0xe9520c24,
+    0x80459473, 0x30300658, 0xa3805ad8, 0xb47eafd2,
+    0x04375a5f, 0x06092cf1, 0x6894ac58, 0xfdfec525,
+    0x0d0889b2, 0x1b804608, 0x1f1f142c, 0xf87ba8c9,
+    0x26800347, 0xf26f3d19, 0xe558e220, 0x145dfa14,
+    0xba317016, 0x3bb06f65, 0x07cf7309, 0xa429c442,
+    0xcce3604d, 0x727f694c, 0x28371b89, 0xd86d80a6,
+    0x2c77943e, 0x60cc8df9, 0x84e9e09c, 0xf31b7067,
+    0xd0158393, 0x29bc8cd8, 0xd88bc6a1, 0xa33f98a8,
+    0xf2340773, 0xab695f5f, 0x28c4f872, 0xcd486133,
+    0x167f7cdf, 0x531faf43, 0xa0847374, 0xd5004438,
+    0xbf6e3c3c, 0x2e720484, 0xb75eee52, 0x10fef2a7,
+    0x63e38812, 0xba96765b, 0x8a244e41, 0xa6120d37,
+    0x15154a10, 0x117869f5, 0x199f5d1c, 0xd6a89007,
+    0x08182d25, 0x48151314, 0xbef97a60, 0xa9f8c8e8,
+    0xf5b1d037, 0x913062cd, 0x8b1d5f46, 0x81eee004,
+    0x12f85444, 0xafb969ab, 0xde74a38f, 0x1287118d,
+    0x8c654586, 0xd0c0a92c, 0x42c7e3b6, 0x6c59f1de,
+    0xc7c22556, 0x1aca992f, 0x0583a104, 0x424053a5,
+    0x53232a5d, 0x2e9907f8, 0x9031a20b, 0x146267d7,
+    0x8af06791, 0x4abffacd, 0x688f40e6, 0x80e8d8f4,
+    0x976b02b3, 0x33de017b, 0x479c8943, 0xd8a744cb,
+    0x412c7311, 0x2b1dee02, 0x649429ba, 0x191d3111,
+    0x041743de, 0x0c67c2a4, 0x4140e20f, 0x350b9685,
+    0xc7c2eea9, 0x16847488, 0xb3027037, 0x38c81979,
+    0x166164a7, 0x7198e5a1, 0x0414241b, 0xb92b29bf,
+    0xe50dbeb4, 0xb711c032, 0xbf398b73, 0x0abed4a6,
+    0x9a048545, 0x94cdafe3, 0x54659149, 0x20ec4cdf,
+    0x6d0a8b14, 0x271b812f, 0xdbee2ab6, 0x811a8a74,
+    0xc5449582, 0x855cdd1e, 0x12b06e17, 0x345a572b,
+    0xbad98795, 0xcdc7038c, 0x1b8d0e61, 0x000000a5,
+    0x00000000, 0x00000000, 0x00000000, 0x00000000,
+    0x00000000, 0x00000000, 0x00000000, 0x00000000,
+    0x00000000, 0x00000000, 0x00000000, 0x00000000,
+};
+
+/* hardcoded here without a bitstream parser helper
+ */
+static VAPictureParameterBufferH264 pic_param[2] ={
+  {
+      CurrPic:{
+          0, 0, 8, 0, 0
+      },
+      ReferenceFrames:{
+          {4294967295, 0, 1, 0, 0},
```

Comment:
use VA_PICTURE_H264_INVALID, VA_INVALID_SURFACE, these definition in va is more suitable than the constant